core/domain/summary_services.py

```diff
@@ -63,6 +63,16 @@ def get_displayable_exp_summary_dicts_matching_ids(exploration_ids):
     displayable_exp_summaries = []
     exploration_summaries = (
         exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+    displayable_exp_summaries = get_displayable_exp_summary_dicts(
+        exploration_summaries, exploration_ids)
+
+    return displayable_exp_summaries
+
+def get_displayable_exp_summary_dicts(exploration_summaries, exploration_ids):
```

Comment:
There is a new issue with this approach. When initially no explorations are loaded then some the exploration_summaries are nonetype while some are empty. If the object is nonetype then an error is observed because it does not has a id property. This can be overcome by checking if each exploration summary is a nonetype.